### PR TITLE
 ✨ Implement external task

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

1. Add `EventAggregator` and `ExternalTaskRepository` to the ServiceTask handler, so it can create ExternalTasks.
2. Add parsing for BPMN tags `camunda:topic` and `camunda:type` to the ServiceTask parser.
3. Remove `WebServiceInvocation` parsing from ServiceTask parser.
4. Split the ServiceTask handler execution function into two main private functions:
  - `_executeInternalServiceTask`: This function handles the ServiceTask as we already know it.
  - `_executeExternalSeviceTask`: This function will use the `ExternalTaskRepository` to create a new ExternalTask.
After that, the ServiceTask gets suspended and waits for a notification about the task having been finished by an external worker.
The handler will only resume, after such a notification has been received.
5. Some general styling

PR: #180

## How can others test the changes?

Use the ProcessEngine to run exernal ServiceTasks.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

**Comment by ElRaptorus** [comment]
https://github.com/process-engine/process_engine_runtime/issues/89

File: src/model/parser/activity_parser.ts [hunk 1]
```diff
@@ -280,25 +286,30 @@ function getMethodInvocationForServiceTask(extensionProperties: Array<Model.Base
...
 }
 
-function getServiceInvocationForServiceTask(extensionProperties: Array<Model.Base.CamundaExtensionProperty>): Model.Activities.WebServiceInvocation {
+function getExternalTaskInvocation(extensionProperties: Array<Model.Base.CamundaExtensionProperty>): Model.Activities.ExternalTaskInvocation {
 
-  const serviceInvocation: Model.Activities.WebServiceInvocation = new Model.Activities.WebServiceInvocation();
+  const externalTaskInvocation: Model.Activities.ExternalTaskInvocation = new Model.Activities.ExternalTaskInvocation();
 
-  const moduleProperty: Model.Base.CamundaExtensionProperty = findExtensionPropertyByName('service', extensionProperties);
+  const urlProperty: Model.Base.CamundaExtensionProperty = findExtensionPropertyByName('url', extensionProperties);
```
Comment (roblen45) [inline src/model/parser/activity_parser.ts]: Hier brauche wir nur Topic und Payload (optional).

Siehe PR von ProcessEngine Contracts

File: src/runtime/engine/flow_node_handler/service_task_handler.ts [hunk 2]
```diff
@@ -37,38 +53,136 @@ export class ServiceTaskHandler extends FlowNodeHandler<Model.Activities.Service
...
+   *
+   * @async
+   * @param   serviceTask        The ServiceTask to execute.
+   * @param   token              The current ProcessToken.
+   * @param   processTokenFacade The Facade for accessing all ProcessTokens of the
+   *                             currently running ProcessInstance.
+   * @param   identity           The identity that started the ProcessInstance.
+   * @returns                    The ServiceTask's result.
+   */
+  private async _executeInternally(serviceTask: Model.Activities.ServiceTask,
```
Comment (roblen45) [inline src/runtime/engine/flow_node_handler/service_task_handler.ts]: Methode umbennen: _executeServiceTask

File: src/runtime/engine/flow_node_handler/service_task_handler.ts [hunk 3]
```diff
@@ -37,38 +53,136 @@ export class ServiceTaskHandler extends FlowNodeHandler<Model.Activities.Service
...
+   * Creates a new ExternalTask and delegates its execution to an
+   * external Service.
+   * The handler will be suspended, until the ExternalTask has finished.
+   *
+   * @async
+   * @param   serviceTask The ServiceTask to execute.
+   * @param   token       The current ProcessToken.
+   * @returns             The ServiceTask's result.
+   */
+  private async _executeExternally(serviceTask: Model.Activities.ServiceTask,
```
Comment (roblen45) [inline src/runtime/engine/flow_node_handler/service_task_handler.ts]: Methode umbenennen: _executeExternalServiceTask

File: src/runtime/engine/flow_node_handler/service_task_handler.ts [hunk 4]
```diff
@@ -37,38 +53,136 @@ export class ServiceTaskHandler extends FlowNodeHandler<Model.Activities.Service
...
+        logger.verbose('External processing of the ServiceTask finished successfully.');
+        resolve(message.result);
+      };
+
+      const subscription: ISubscription = this._eventAggregator.subscribeOnce(externalTaskFinishedEventName, messageReceivedCallback);
+
+      logger.verbose('Persist ServiceTask as ExternalTask.');
+      await this
+        ._externalTaskRepository
+        .create(serviceTask.topic, token.correlationId, token.processInstanceId, this.flowNodeInstanceId, serviceTask.invocation);
```
Comment (roblen45) [inline src/runtime/engine/flow_node_handler/service_task_handler.ts]: Invocation muss raus und er payload raus

**Pull request merged**